firmware/common/i2cdev_max17055.cpp

```diff
@@ -184,11 +184,24 @@ const RegisterEntry MAX17055::entries[] = {
     {"Reserved", 0xFF, "", 0.00390625, false, "%", true, 6, true, false, false, 0, true},
 };
 
-void MAX17055::init() {
-    if (!detected_) {
-        detected_ = detect();
-    }
-    if (detected_) {
+void I2cDev_MAX17055::update() {
+    uint8_t validity = 0;
+    uint8_t batteryPercentage = 102;
+    uint16_t voltage = 0;
+    int32_t current = 0;
+    getBatteryInfo(validity, batteryPercentage, voltage, current);
+
+    // send local message
+    BatteryStateMessage msg{validity, batteryPercentage, current >= 0, voltage};
+    EventDispatcher::send_message(msg);
+}
+
+bool I2cDev_MAX17055::init(uint8_t addr_) {
```

Comment:
it tries to init the deice (there are the checks), returns true if success, and then the thread will start to poll it.
if false it won't use that. 
it tries to init every new device on detect. like you plug in, it'll try to find a driver. if you unplug and replug, then it'll try it again.

in context of battery ic, it detects the ic, checks the driver, the driver reports it can handle it, and it'll be there till power down.